src/debugCodeLensProvider.ts

```diff
@@ -181,7 +181,9 @@ async function launchJsonExists(workspace?: vscode.Uri): Promise<boolean> {
     }
 
     const workspaceFolder = vscode.workspace.getWorkspaceFolder(workspace);
-    const results: vscode.Uri[] = await vscode.workspace.findFiles(".vscode/launch.json");
+    // Excluding "**/node_modules/**" as a common cause of excessive CPU usage.
+    // https://github.com/microsoft/vscode/issues/75314#issuecomment-503195666
+    const results: vscode.Uri[] = await vscode.workspace.findFiles(".vscode/launch.json", "**/node_modules/**");
```

Comment:
I'm OK with excluding both `node_modules` and `.git`.

Reply:
Thanks! So to be clear, updating this argument `**/node_modules/**` to `{**/node_modules/**,.git/**}` would be ideal?

I'll test that out to make sure I have the right glob syntax.

Reply:
It looks like the 2nd argument appends rather than replaces. The existing command in this PR:

```ts
await vscode.workspace.findFiles(".vscode/launch.json", "**/node_modules/**");
```

runs the following process:

```
'/Applications/Visual Studio Code.app/Contents/Resources/app/node_modules.asar.unpacked/@vscode/ripgrep/bin/rg' \
    --files \
    --hidden \
    --case-sensitive \
    -g '/.vscode/launch.json' \
    -g '!**/node_modules/**' \
    -g '!**/.git' \
    -g '!**/.svn' \
    -g '!**/.hg' \
    -g '!**/CVS' \
    -g '!**/.DS_Store' \
    -g '!**/Thumbs.db' \
    --no-ignore \
    --follow \
    --no-config \
```

The `-g '!**/.git'` exclusion comes from [VS Code's default settings](https://code.visualstudio.com/docs/getstarted/settings#_default-settings).

```json
{
  "files.exclude": {
    "**/.git": true,
    "**/.svn": true,
    "**/.hg": true,
    "**/CVS": true,
    "**/.DS_Store": true,
    "**/Thumbs.db": true
  }
}
```

https://github.com/microsoft/vscode/blob/2798da9d47387c0fb6ab4ebbead95efb5a85c250/src/vs/workbench/contrib/files/browser/files.contribution.ts#L143

> It would be interesting if there is some performance comparison data to share.

Definitely. I was curious too, but hadn't had time until now to run some tests.

Without the `**/node_modules/**` exclusion, the `rg` process runs for over an hour at 1100%+ CPU in the repo my team works on. I can run the process overnight to see exactly how long it takes if that's something we're curious about.

Here's some data for comparing whether `.git` is excluded. For my specific machine and repo:

|`node_modules` Excluded|`.git` Excluded|Trial 1|Trial 2|Average|
|-|-|-|-|-|
|❌|✅|1 hour+|-|1 hour+|
|✅|❌|536ms|493ms|514.5ms|
|✅|✅|788ms|477ms|632.5ms|

It looks like excluding `.git` is fairly negligible in terms of timing. Either way, I don't think we have to do something explicit here since `.git` should be excluded unless users explicitly set `files.excludes["**/.git"]` to `false`?

Reply:
Great to see the perf data, thanks for sharing that. Looks like it's enough to exclude "node_modules" only.

Reply:
Looks like this utility can get another improvement if we set the `workspaceFolder` as the base part of the relative pattern

https://github.com/microsoft/vscode/blob/4b759b0492d1df969e8595e3db0f66f6058c3643/src/vscode-dts/vscode.d.ts#L2116

And then limit the search maxResults to 1

https://github.com/microsoft/vscode/blob/4b759b0492d1df969e8595e3db0f66f6058c3643/src/vscode-dts/vscode.d.ts#L12171

Reply:
I created an issue to track the further optimization. https://github.com/microsoft/vscode-java-debug/issues/1252

Reply:
@jdneo Those both look like great suggestions. Thanks!